src/Angor/Client/Services/MessageService.cs

```diff
@@ -0,0 +1,393 @@
+                    async eventMessage => await ProcessDirectMessage(eventMessage),
+                    _currentUserHexPub
+                );
+
+                await _relayService.LookupDirectMessagesForPubKeyAsync(
+                    _currentUserHexPub,
+                    DateTime.UtcNow.AddDays(-7),
+                    100,
+                    async eventMessage => await ProcessDirectMessage(eventMessage),
+                    _contactHexPub
+                );
+
+                _directMessages = _directMessages.OrderBy(m => m.Timestamp).ToList();
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError($"Failed to load messages: {ex.Message}");
+            }
+            finally
+            {
+                NotifyStateChanged(); 
+            }
+        }
+
+        private async Task LoadNewMessagesAsync()
+        {
+            if (string.IsNullOrEmpty(_currentUserPrivateKeyHex) || string.IsNullOrEmpty(_contactHexPub) || string.IsNullOrEmpty(_currentUserHexPub))
+            {
+                _logger.LogWarning("LoadNewMessagesAsync - Missing required keys.");
+                return;
+            }
+
+            DateTime sinceTime = DateTime.UtcNow.AddDays(-7);
+            if (_directMessages.Any())
+            {
+                sinceTime = _directMessages.Max(m => m.Timestamp);
+            }
+
+            try
+            {
+                await _relayService.LookupDirectMessagesForPubKeyAsync(
+                    _currentUserHexPub,
+                    sinceTime,
+                    100,
+                    async eventMessage => await ProcessDirectMessage(eventMessage),
+                    _contactHexPub
+                );
+
+                await _relayService.LookupDirectMessagesForPubKeyAsync(
+                    _contactHexPub,
+                    sinceTime,
+                    100,
+                    async eventMessage => await ProcessDirectMessage(eventMessage),
+                    _currentUserHexPub
+                );
+
+                _directMessages = _directMessages.OrderBy(m => m.Timestamp).ToList();
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError($"Failed to load new messages: {ex.Message}");
+            }
+            finally
+            {
+                NotifyStateChanged();
+            }
+        }
+
+        private void EnsureActiveSubscription()
+        {
+            if (_subscriptionActive)
+            {
+                _logger.LogDebug("Message subscription already active, no need to resubscribe");
+                return;
+            }
+
+            if (string.IsNullOrEmpty(_currentUserHexPub) || string.IsNullOrEmpty(_contactHexPub))
+            {
+                _logger.LogWarning("EnsureActiveSubscription - Missing required keys.");
+                return;
+            }
+
+            try
+            {
+                _messageSubscription?.Dispose();
```

Comment:
its odd that the method `EnsureActiveSubscription` will destroy the subscription and recreate it no?

Reply:
but dispose will clean up everything even valid state subscriptions no?
I mean it is probably ok but inefficient